web/src/engine/osk/src/banner/suggestionBanner.ts

```diff
@@ -170,7 +176,22 @@ export class BannerSuggestion {
 
     if(suggestion && suggestion.displayAs) {
       const rawMetrics = getTextMetrics(suggestion.displayAs, format.emSize, format.styleForFont);
-      this._textWidth = rawMetrics.width;
+      let projectedHeight = 0;
+      if(rawMetrics.fontBoundingBoxAscent) {
+        projectedHeight = rawMetrics.fontBoundingBoxAscent + rawMetrics.fontBoundingBoxDescent;
+      }
+      let ratio = Math.min(1, format.height / projectedHeight);
```

Comment:
This could lead to a divide by zero, which isn't a problem in Javascript, but IMO has a bit of a smell. Why not initialize `projectedHeight = 1` instead of `0`?